### PR TITLE
MAP-1770: Fixing non-res room type error

### PR DESCRIPTION
**Pull request by ashleygyngell**
Fix where validation error clears selection of "Other" type when description is not entered and save is attempted for non-residential room.

### CODE REVIEW

File: server/controllers/changeNonResidentialType/details.ts [hunk 1]
```diff
@@ -44,6 +45,29 @@ export default class ChangeNonResidentialTypeDetails extends FormInitialStep {
...
+      if (
+        (convertedCellTypeUnchanged && convertedCellType !== 'OTHER') ||
+        (convertedCellType === 'OTHER' && convertedCellTypeUnchanged && otherConvertedCellTypeUnchanged)
+      ) {
+        return res.redirect(`/view-and-update-locations/${prisonId}/${locationId}`)
+      }
+      return callback({ ...errors })
+    })
+  }
+
```
Comment (ashleygyngell) [inline server/controllers/changeNonResidentialType/details.ts:70]: Alongside returning errors, validation redirects if no changes have been made. If changes have, we show success banner.

**Review by Mjwillis — APPROVED** [review]
LGTM

File: server/controllers/changeNonResidentialType/details.ts [hunk 2]
```diff
@@ -28,14 +28,15 @@ export default class ChangeNonResidentialTypeDetails extends FormInitialStep {
...
-
-    const convertedCellTypes = location.raw?.convertedCellType || []
+    const convertedCellType = req.form.values.convertedCellType || location.raw?.convertedCellType || ''
+    const otherConvertedCellType = req.form.values.otherConvertedCellType || location.raw?.otherConvertedCellType || ''
 
     fields.convertedCellType.items = fields.convertedCellType.items.map((item: FormWizard.Field) => ({
       ...item,
-      checked: convertedCellTypes.includes(item.value), // Use the new variable
+      checked: item.value === convertedCellType,
     }))
```
Comment (ashleygyngell) [inline server/controllers/changeNonResidentialType/details.ts:37]: Makes sure the correct item is checked. Was reverting before when selecting the 'other' field.

File: server/controllers/changeNonResidentialType/details.ts [hunk 3]
```diff
@@ -28,14 +28,15 @@ export default class ChangeNonResidentialTypeDetails extends FormInitialStep {
...
     const fields = { ...locals.fields }
-
-    const convertedCellTypes = location.raw?.convertedCellType || []
+    const convertedCellType = req.form.values.convertedCellType || location.raw?.convertedCellType || ''
+    const otherConvertedCellType = req.form.values.otherConvertedCellType || location.raw?.otherConvertedCellType || ''
 
     fields.convertedCellType.items = fields.convertedCellType.items.map((item: FormWizard.Field) => ({
       ...item,
-      checked: convertedCellTypes.includes(item.value), // Use the new variable
+      checked: item.value === convertedCellType,
```
Comment (ashleygyngell) [inline server/controllers/changeNonResidentialType/details.ts:36]: Makes sure the correct item is checked. Was reverting before when selecting the 'other' field.

**Pull request merged**